### PR TITLE
Basics Features and Tests

### PR DESCRIPTION
**Pull request by guilyx**
## Purpose
Introduce a utility library for geometry stuff, as some things have been reused in multiple places. Trying to reduce code duplication.

## Approach
This will add some distance calculation helper (angle to angle, point to point, pose to pose, quaternion to quaternion etc etc), as well as some tf helpers, and more if possible.

## Disclaimers
This PR doesn't account for all that should be in there and the package should be populated by anyone who thinks it should be in there.
Ideally we make a Python version
-> use the same repo, add two packages, `${package_name}_py` and `${package_name}_cpp`

### CODE REVIEW

**Comment by guilyx** [comment]
@SomaGallai

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+include_directories(include
+  ${rclcpp_INCLUDE_DIRS}
+  ${geometry_msgs_INCLUDE_DIRS}
+  ${tf2_ros_INCLUDE_DIRS}
+  ${tf2_INCLUDE_DIRS}
+  ${tf2_geometry_msgs_INCLUDE_DIRS}
+  ${angles_INCLUDE_DIRS}
+)
```
Comment (cmraaron) [inline CMakeLists.txt:34]: shouldnt need this if you use `ament_target_dependencies`

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+target_link_libraries(${LIBRARY_NAME} INTERFACE
+  ${rclcpp_LIBRARIES}
+  ${geometry_msgs_LIBRARIES}
+  ${tf2_ros_LIBRARIES}
+  ${tf2_geometry_msgs_LIBRARIES}
+  ${tf2_LIBRARIES}
+  ${angles_LIBRARIES}
+)
```
Comment (cmraaron) [inline CMakeLists.txt:45]: dont need these old style variable expansion.  and probably want to use the `ament` version of this for ros libs

File: README.md [hunk 3]
```diff
@@ -1,2 +1,49 @@
-# cmr_geometry_utils
-CMR geometry utils library
+<div align="center">
+
+# CMR Tests Utils
```
Comment (cmraaron) [inline README.md]: but the repo is called `cmr_geometry_utils`?  confused

File: package.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  <version>0.0.1</version>
+  <description>A header only library to facilitate writing gtest unit tests for ROS2 nodes.</description>
+  <maintainer email="erwin.lejeune@cm-robotics.com">ros</maintainer>
+  <license>Apache License 2.0</license>
+  
+  <build_depend>geometry_msgs</build_depend>
+  <build_depend>tf2_ros</build_depend>
+  <build_depend>tf2</build_depend>
+  <build_depend>tf2_geometry_msgs</build_depend>
+  <build_depend>angles</build_depend>
```
Comment (cmraaron) [inline package.xml]: are these not runtime deps also?

File: include/cmr_geometry_utils/basic_geometry_utils.hpp [hunk 5]
```diff
@@ -7,17 +7,45 @@
...
-double compute_angle_distance(const geometry_msgs::msg::Quaternion & start, const geometry_msgs::msg::Quaternion & goal)
+double compute_distance_2d(const geometry_msgs::msg::Pose & start, const geometry_msgs::msg::Pose & goal)
+{
+  return std::hypot(goal.position.x - start.position.x, 
+                    goal.position.y - start.position.y);
+}
+
+double compute_distance_2d(const geometry_msgs::msg::PoseStamped & start, const geometry_msgs::msg::PoseStamped & goal)
+{
+  return std::hypot(goal.pose.position.x - start.pose.position.x,
```
Comment (cmraaron) [inline include/cmr_geometry_utils/basic_geometry_utils.hpp]: perhaps just define this overload in terms of the prior one.  ie pass `goal.pose` and `start.pose` to the first `compute_distance_2d`

File: include/cmr_geometry_utils/basic_geometry_utils.hpp [hunk 6]
```diff
@@ -7,17 +7,45 @@
...
+double compute_yaw_distance(const geometry_msgs::msg::Pose & start, const geometry_msgs::msg::Pose & goal)
+{
+  double start_yaw = angles::normalize_angle(tf2::getYaw(start.orientation));
+  double goal_yaw = angles::normalize_angle(tf2::getYaw(goal.orientation));
+  return std::abs(angles::normalize_angle(goal_yaw - start_yaw));
+}
+
+double compute_yaw_distance(const geometry_msgs::msg::PoseStamped & start, const geometry_msgs::msg::PoseStamped & goal)
+{
+  double start_yaw = angles::normalize_angle(tf2::getYaw(start.pose.orientation));
```
Comment (cmraaron) [inline include/cmr_geometry_utils/basic_geometry_utils.hpp]: same here, define the overloaded `compute_yaw_distance` in terms of the first

**Review by cmraaron — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (guilyx) [reply README.md]: oops readme isn't updated

**Review by cmraaron — DISMISSED** [review]

**Comment by SomaGallai** [comment]
@cmraaron maybe you could help me with the CI. The following issue occurs during the Debian creation: `/usr/bin/ld: cannot find -lpthreads`

**Comment by cmraaron** [comment]
> @cmraaron maybe you could help me with the CI. The following issue occurs during the Debian creation: `/usr/bin/ld: cannot find -lpthreads`

no problem.  was a bit of a red herring this one

**Comment by guilyx** [comment]
@SomaGallai @cmraaron ci passing :-)

**Review by cmraaron — APPROVED** [review]

**Comment by guilyx** [comment]
@SomaGallai you took too long homeboy

**Pull request merged**